source/Engine/Bytecode/StandardLibrary.cpp

```
@@ -9783,54 +9783,36 @@ VMValue Model_DeleteArmature(int argCount, VMValue* args, Uint32 threadID) {
-	}
-	if (audio) {
-		AudioManager::PushMusicAt(
-			audio, 0.0, false, 0, panning, speed, volume, fadeInAfterFinished);
-	}
-	return NULL_VAL;
-}
-/***
- * Music.PlayAtTime
- * \desc Places the music onto the music stack and plays it at a time (in seconds).
- * \param music (Integer): The music index to play.
- * \param startPoint (Decimal): The time (in seconds) to start the music at.
- * \paramOpt panning (Decimal): Control the panning of the audio. -1.0 makes it sound in left ear only, 1.0 makes it sound in right ear, and closer to 0.0 centers it. (0.0 is the default.)
- * \paramOpt speed (Decimal): Control the speed of the audio. > 1.0 makes it faster, < 1.0 is slower, 1.0 is normal speed. (1.0 is the default.)
- * \paramOpt volume (Decimal): Controls the volume of the audio. 0.0 is muted, 1.0 is normal volume. (1.0 is the default.)
- * \paramOpt fadeInAfterFinished (Decimal): The time period to fade in the previous music track after the currently playing track finishes playing, in seconds. (0.0 disables this.)
- * \ns Music
- */
-VMValue Music_PlayAtTime(int argCount, VMValue* args, Uint32 threadID) {
-	CHECK_AT_LEAST_ARGCOUNT(2);
-	ISound* audio = GET_ARG(0, GetMusic);
-	double start_point = GET_ARG(1, GetDecimal);
+	int loopPoint = IS_NULL(args[1]) ? MUSIC_LOOP_DEFAULT : GET_ARG_OPT(1, GetInteger, MUSIC_LOOP_DEFAULT);
 	float panning = GET_ARG_OPT(2, GetDecimal, 0.0f);
 	float speed = GET_ARG_OPT(3, GetDecimal, 1.0f);
 	float volume = GET_ARG_OPT(4, GetDecimal, 1.0f);
-	float fadeInAfterFinished = GET_ARG_OPT(5, GetDecimal, 0.0f);
-	if (fadeInAfterFinished < 0.f) {
+	double startPoint = GET_ARG_OPT(5, GetDecimal, 0.0);
+	float fadeInAfterFinished = GET_ARG_OPT(6, GetDecimal, 0.0f);
+
+	if (fadeInAfterFinished < 0.f)
 		fadeInAfterFinished = 0.f;
-	}
-	if (audio) {
-		AudioManager::PushMusicAt(
-			audio, start_point, false, 0, panning, speed, volume, fadeInAfterFinished);
-	}
+
+	if (loopPoint < MUSIC_LOOP_NONE)
+		loopPoint = MUSIC_LOOP_NONE;
```

Might be better to throw an error here